src/plinder/data/splits.py

```diff
@@ -826,3 +826,108 @@ def split(*, data_dir: Path, cfg: DictConfig, relpath: str) -> pd.DataFrame:
         system_to_cluster_val,
         split_path,
     )
+
+
+def sort_ccd_codes(ccd_codes_str: str) -> str:
+    return "-".join(sorted(ccd_codes_str.split("-")))
+
+
+def ccd_codes_to_set(ccd_codes_str: str) -> Set[str]:
+    return set(ccd_codes_str.split("-"))
+
+
+def map_pdb_ccd_code_to_system_ids(
+    data_dir: Path, pdbid_ccd_codes: Set[str]
+) -> Tuple[Dict[str, str], Set[str]]:
+    annotations = pd.read_parquet(data_dir / "index/annotation_table.parquet")
+    pdbid_ccd_codes = {i.upper() for i in pdbid_ccd_codes}
+
+    # VO: not sure if we need this here
+    # annotations["ligand_ccd_code"] = annotations["ligand_ccd_code"].apply(
+    #     lambda x: sort_ccd_codes(x)
```

Comment:
Not sure if sorting code is relevant here as posebusters is using only one ligand codes. I thus wrote an "exploded" comparison instead. See below!